magenta-lib/src/main/scala/magenta/json/JsonInputFile.scala

```diff
@@ -38,24 +43,26 @@ case class JsonRecipe(
 object JsonReader {
   private implicit val formats = DefaultFormats
 
-  def parse(f: File)(implicit reporter: DeployReporter): Project = {
+  def parse(artifact: S3Artifact)(implicit s3Client: AmazonS3Client, reporter: DeployReporter): Project = {
     try {
-      parse(Source.fromFile(f).mkString, f.getAbsoluteFile.getParentFile)
+      val deployJson = s3Client.getObject(artifact.bucket, s"${artifact.key}/${artifact.deployObject}")
+      val deployJsonInputStream = deployJson.getObjectContent
+      parse(Source.fromInputStream(deployJsonInputStream).mkString, artifact)
```

Comment:
Thanks Chris - good spot. Actually found a couple of other places this was done. Have fixed them all.